### PR TITLE
[feat][pat] add Add Folder button and modal

### PR DESCRIPTION
**Pull request by PatTumulak**
**What this PR does:**
- added an Add Folder button
- added an Add Folder modal

**Screenshots: **
Screen Shot 2021-08-09 at 2 30 33 PM
Screen Shot 2021-08-09 at 4 55 00 PM

### CODE REVIEW

File: src/components/AddFolderButton/index.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import { Button } from 'antd';
+import { FolderAddFilled } from '@ant-design/icons';
+
+import AddFolderModal from 'components/AddFolderModal';
+
+import './styles.scss';
+
+function AddFolderButton() {
+  const [isAddFolderModalVisible, setIsAddFolderModalVisible] = useState(false);
```
Comment (itskichaem) [inline src/components/AddFolderButton/index.jsx]: lets simplify this to `isModalVisible`

File: src/components/AddFolderModal/index.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+import { Button, Form, Input, Modal } from 'antd';
+
+import './styles.scss';
+
+function AddFolderModal({ handleCancel, isModalVisible }) {
```
Comment (itskichaem) [inline src/components/AddFolderModal/index.jsx]: event function props should be named `on-`. on this case this should be `onCancel`. 

more on that here https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

File: src/components/AddFolderButton/index.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  return (
+    <>
+      <AddFolderModal
+        handleCancel={() => setIsAddFolderModalVisible(false)}
```
Comment (itskichaem) [inline src/components/AddFolderButton/index.jsx]: lets avoid inlining functions. there should be a `handleCancel`  fn that does this.

File: src/components/AddFolderButton/index.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        isModalVisible={isAddFolderModalVisible}
+      />
+      <Button
+        className="dashboard-folder-button"
+        icon={<FolderAddFilled />}
+        onClick={() => setIsAddFolderModalVisible(true)}
```
Comment (itskichaem) [inline src/components/AddFolderButton/index.jsx]: lets avoid inlining functions. there should be a `handleOpenModal`  fn that does this.

**Review by itskichaem — CHANGES_REQUESTED** [review]

File: src/components/AddFolderModal/styles.scss [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    border-top-right-radius: 16px;
+  }
+
+  .ant-input-lg {
+    font-size: 22px;
+    color: $black;
+  }
+
+  .ant-btn-primary {
+    color: #fff;
```
Comment (AnthonyDelaCruz) [inline src/components/AddFolderModal/styles.scss]: use reusable colors

File: src/components/AddFolderModal/index.jsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  return (
+    <Modal
+      className="modal"
+      destroyOnClose="true"
+      footer={null}
```
Comment (AnthonyDelaCruz) [inline src/components/AddFolderModal/index.jsx]: if it's null we can just omit the prop altogether.

File: src/components/AddFolderModal/index.jsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+import { Button, Form, Input, Modal } from 'antd';
+
+import './styles.scss';
+
+function AddFolderModal({ handleCancel, isModalVisible }) {
+  return (
+    <Modal
+      className="modal"
+      destroyOnClose="true"
```
Comment (AnthonyDelaCruz) [inline src/components/AddFolderModal/index.jsx]: I think you are passing the wrong prop type here for `destroyClose`, the Modal API in the docs says it accepts a boolean not a string, but I understand the intention, also If the prop is always true, we don't have to explicitly pass `true` as a value. We can just place the prop `destroyOnClose` without assigning a value. This means that it's always true.

**Review by AnthonyDelaCruz — CHANGES_REQUESTED** [review]

**Review by itskichaem — APPROVED** [review]

**Pull request merged**